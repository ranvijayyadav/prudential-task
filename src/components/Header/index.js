import "./index.scss";
import React, { PureComponent } from "react";
import {
    AppBar,    Button
} from "@material-ui/core";
import logo from "./../../assets/images/logo_white_small.png";
export default class Header extends PureComponent {
    render() {

        return (
            <div>
                <AppBar position="fixed" color="inherit" className="headerbar">
                    
                    <Button href="#" className="buttonw">
                    <img src={logo} alt="logo" className="logo" />
                        <p>WEATHER APP</p>
                    </Button>
                </AppBar>
            </div>
        );
    }
}
