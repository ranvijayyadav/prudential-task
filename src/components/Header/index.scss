@import "assets/scss/variables.scss";

.buttonw {
    width: 60px;
    background-color: $primary;
    border-radius: 0;
    float: left;
    height: 60px;
    display: block;
    min-height: 60px;
}

.logo {
    width: 35px;
}

.menulinks {
    float: left;
    margin: 5px 0 0 10px;
}

.listmenu {
    float: left;
    display: block;
    width: auto;
    padding: 5px 5px 5px 20px;
    margin: 0 5px;

    > div {
        padding: 0;
    }
}


.headerbar {
    display: block;
    height: 60px;
}

.userwrap {
    float: right;

    .saveBtn {
        display: inline-block;
        vertical-align: middle;
        margin: 0 20px 0 0;
    }

    .bigAvatar {
        border-radius: 0;
        width: auto;
        height: 60px;
        padding: 10px 15px;
        background: lightgray;
        color: black;
        cursor: pointer;
        display: inline-block;
        vertical-align: middle;
    }

    img,
    span {
        display: inline-block;
        vertical-align: middle;
    }

    img {
        max-width: 40px;
    }

    span {
        padding: 0 0 0 5px;
        font-size: 14px;
    }

    img.icon {
        width: 24px;
        padding: 0 0 0 10px;
    }
}
