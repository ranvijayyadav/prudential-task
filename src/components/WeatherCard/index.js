import React from 'react';
import environments from "./../../environments";
const WeatherCard=(props)=>{
    const tile = props.tile;
    const weatherIcon= `${environments.IMAGE_URL}${tile.weather[0].icon}.png`
    
return(
    <div>
        <h5>Time: {tile.dt_txt}</h5>
        <p>
            <span>General</span>
            <strong><span> {String(tile.weather[0].description).toLocaleUpperCase()}</span></strong>
        </p>   
        <img src={weatherIcon} alt="Smiley face" height="42" width="42"/>
        <ul>
            <li>Temp:{convertToCelcious(tile.main.temp)}°</li>
            <li>Max Temp:{convertToCelcious(tile.main.temp_max)}°</li>
            <li>Min Temp:{convertToCelcious(tile.main.temp_min)}°</li>
            <li>Wind Speed:{tile.wind.speed} mph</li>
            <li>Humidity:{tile.main.humidity}%</li>
        </ul>
    </div>)

}
const convertToCelcious=(temp)=>{
    return parseInt((temp - 273.15), 10)
}

export default WeatherCard;