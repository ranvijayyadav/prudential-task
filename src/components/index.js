export { default as Header } from "./Header";
export {default as WeatherList} from "./WeatherList";
export {default as WeatherCard} from "./WeatherCard";

