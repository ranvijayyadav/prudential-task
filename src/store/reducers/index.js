import { combineReducers } from "redux";
import { default as weather } from "./weather";

export default combineReducers({
    weather
});
