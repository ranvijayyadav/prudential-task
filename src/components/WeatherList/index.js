import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import ListSubheader from '@material-ui/core/ListSubheader';
import {connect} from "react-redux";
import {WeatherCard} from "./../../components";
import Loading from "./../../assets/images/spinner.gif";

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper,
  },
  gridList: {
    margin:"80px !important",
    width: "65%",
    height: "100%",
  },
  gridItem:{
    border: "1px solid peru",
    height: "250px !important",
    margin: "1px, 1px, 1px, 1px"
  },
  loader:{
    marginTop:"200px",
    marginLeft:"500px"
  },
  icon: {
    color: 'rgba(255, 255, 255, 0.54)',
  },
});


function WeatherList(props) {
  const { classes } = props;
if(props.weather.loaded){
  const tileData = props.weather.data;
  const monthNames = ["January", "February", "March", "April", "May", "June",
  "July", "August", "September", "October", "November", "December"
];
  const date= new Date();
debugger;
return (
    <div className={classes.root}>
      <GridList cellHeight={180} className={classes.gridList}>
        <GridListTile key="Subheader" cols={2} style={{ height: 'auto' }}>
          <ListSubheader component="div">{tileData.city.name} : {monthNames[date.getMonth()]}</ListSubheader>
        </GridListTile>
        {tileData.list.map(tile => (
          <GridListTile className={classes.gridItem} key={tile.img}>
          <WeatherCard tile={tile} classes={classes}/>
          </GridListTile>
          
        ))}
      </GridList>
    </div>
  );
} else{
  return(<div className={classes.loader}>
    
    <img src={Loading} alt="Loading ..."/>
  </div>)
}
}

WeatherList.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  weather: state.weather
})
export default connect(mapStateToProps)(withStyles(styles)(WeatherList));