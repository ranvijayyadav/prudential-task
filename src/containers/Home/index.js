import React, { Component } from "react";
import { create } from "jss";
import JssProvider from "react-jss/lib/JssProvider";
import {
    createGenerateClassName,
    jssPreset,
    CssBaseline
} from "@material-ui/core";
import { Header} from "./../../components";
import { Main } from "./../../containers";

const styleNode = document.createComment("insertion-point-jss");
document.head.insertBefore(styleNode, document.head.firstChild);

const generateClassName = createGenerateClassName();
const jss = create(jssPreset());
jss.options.insertionPoint = "insertion-point-jss";

class Home extends Component {
    render() {
        return (
            <JssProvider jss={jss} generateClassName={generateClassName}>
                <React.Fragment>
                    <CssBaseline />
                    <Header />
                    <Main />
                </React.Fragment>
            </JssProvider>
        );
    }
}

export default Home;
