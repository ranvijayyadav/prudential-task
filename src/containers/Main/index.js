import "./index.scss";
import React, { Component } from "react";
import {WeatherList} from "./../../components";
import {connect} from "react-redux";
import {getWeather} from "./../../store";
//import "./../../../assets/fonts/stylesheet.scss";

class Main extends Component {
    componentWillMount(){
        console.log("api call");
        this.props.getWeather("Mumbai","IN");
    }
    render() {
        return (
            <div>
                <WeatherList/>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    weather: state.weather
})
//export default Main;
export default connect(mapStateToProps, {getWeather})(Main)