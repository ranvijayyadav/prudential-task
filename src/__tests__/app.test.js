import React from "react";
import {create} from "react-test-renderer";
import {WeatherCard, WeatherList, Header} from "./../components";
import {getWeather} from "./../store";
import moxios from 'moxios';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store'
import {Main} from "./../containers";

import {GET_WEATHER} from "./../store/constants";
const jsondata= {
    "dt": 1537909200,
    "main": {
        "temp": 279.08,
        "temp_min": 279.04,
        "temp_max": 279.08,
        "pressure": 1007.5,
        "sea_level": 1027.56,
        "grnd_level": 1007.5,
        "humidity": 88,
        "temp_kf": 0.04
    },
    "weather": [{
        "id": 803,
        "main": "Clouds",
        "description": "broken clouds",
        "icon": "04n"
    }],
    "clouds": {
        "all": 80
    },
    "wind": {
        "speed": 5.22,
        "deg": 312.5
    },
    "rain": {},
    "sys": {
        "pod": "n"
    },
    "dt_txt": "2018-09-25 21:00:00"
};
test("snapshot weather card", ()=>{
 
    const wc= create(<WeatherCard tile={jsondata}/>);
    expect(wc.toJSON()).toMatchSnapshot();
    
});

test("snapshot Header", ()=>{
 
    const head= create(<Header/>);
    expect(head.toJSON()).toMatchSnapshot();
    
});

test("action creater", ()=>{
    //const store = mockStore({ posts: {} })
    expect(getWeather().type).toBe(GET_WEATHER);
});



test("main component", ()=>{


    // create any initial state needed
    const initialState = {};

    const mockStore = configureStore();

    let wrapper;
    beforeEach(() => {
            store = mockStore(initialState)
            wrapper = wrapper = shallow(<Main store={store}/>)

    })

        it('component is rendered', () => {
            expect(wrapper.length).toEqual(1)
    });
});