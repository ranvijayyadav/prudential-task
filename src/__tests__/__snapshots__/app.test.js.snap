// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Header 1`] = `
<div>
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionFixed-2 mui-fixed headerbar"
  >
    <a
      className="MuiButtonBase-root-63 MuiButton-root-37 MuiButton-text-39 MuiButton-flat-42 buttonw"
      href="#"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-38"
      >
        <img
          alt="logo"
          className="logo"
          src="logo_white_small.png"
        />
        <p>
          WEATHER APP
        </p>
      </span>
      <span
        className="MuiTouchRipple-root-66"
      />
    </a>
  </header>
</div>
`;

exports[`snapshot weather card 1`] = `
<div>
  <h5>
    Time: 
    2018-09-25 21:00:00
  </h5>
  <p>
    <span>
      General
    </span>
    <strong>
      <span>
         
        BROKEN CLOUDS
      </span>
    </strong>
  </p>
  <img
    alt="Smiley face"
    height="42"
    src="http://openweathermap.org/img/w/04n.png"
    width="42"
  />
  <ul>
    <li>
      Temp:
      5
      °
    </li>
    <li>
      Max Temp:
      5
      °
    </li>
    <li>
      Min Temp:
      5
      °
    </li>
    <li>
      Wind Speed:
      5.22
       mph
    </li>
    <li>
      Humidity:
      88
      %
    </li>
  </ul>
</div>
`;
